app/Http/Controllers/UsersController.php

```diff
@@ -85,7 +86,37 @@ public function edit($id)
      */
     public function update(Request $request, $id)
     {
-        //
+        $user = User::find($id);
+
+        // Handle the user upload of avatar
+        if ($request->hasFile('avatar')) {
```

Comment:
create `setter` function in a different `Class` you can declare it in your `Model` Class. Point is that a function should only do one thing. From this code it do image uploading and updating, but we just want it for updating user information. So Image uploading should be set as a different method. and that you can just call that method which you will pass in your update functions